src/Controls/src/Core/Handlers/Items/Android/SpacingItemDecoration.cs

```diff
@@ -56,16 +61,76 @@ public SpacingItemDecoration(Context context, IItemsLayout itemsLayout)
 
 			HorizontalOffset = (int)context.ToPixels(horizontalOffset);
 			VerticalOffset = (int)context.ToPixels(verticalOffset);
+			_itemsLayout = itemsLayout;
 		}
 
 		public override void GetItemOffsets(ARect outRect, AView view, RecyclerView parent, RecyclerView.State state)
 		{
 			base.GetItemOffsets(outRect, view, parent, state);
-
+			int position = parent.GetChildAdapterPosition(view);
+			int itemCount = state.ItemCount;
 			outRect.Left = HorizontalOffset;
 			outRect.Right = HorizontalOffset;
 			outRect.Bottom = VerticalOffset;
 			outRect.Top = VerticalOffset;
+
+
+			if (_itemsLayout is GridItemsLayout gridItemsLayout)
+			{
+				if (gridItemsLayout.Orientation == ItemsLayoutOrientation.Vertical)
+				{
+					int row = position / _span;
+					int totalRows = (int)Math.Ceiling((double)itemCount / _span);
+
+					if (row == 0)
+					{
+						outRect.Top = 0;
+					}
+					else if (row == totalRows - 1)
+					{
+						outRect.Bottom = 0;
+					}
+				}
+				else
+				{
+					int column = position / _span;
+					int totalColumns = (int)Math.Ceiling((double)itemCount / _span);
+
+					if (position < _span)
```

Comment:
This is different with the `<`. Is this intentional? This is a horizontal grid, which means it is a grid of _span rows or all the items in the first column. Which is basically column == 0?